internal/v5/resources_test.go

```diff
@@ -461,3 +461,160 @@ func (s *ResourceSuite) TestListResourceWithBundle(c *gc.C) {
 		},
 	})
 }
+
+func (s *ResourceSuite) TestMetaResourcesSingleBadResourceId(c *gc.C) {
+	id := newResolvedURL("~charmers/precise/wordpress-0", -1)
+	s.addPublicCharm(c, storetesting.NewCharm(nil), id)
+	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
+		Handler:      s.srv,
+		URL:          storeURL(id.URL.Path() + "/meta/resources/foo/bar"),
+		ExpectStatus: http.StatusNotFound,
+		ExpectBody: params.Error{
+			Code:    params.ErrMetadataNotFound,
+			Message: string(params.ErrMetadataNotFound),
+		},
+	})
+}
+
+func (s *ResourceSuite) TestMetaResourcesSingleResourceWithNegativeRevisionNumber(c *gc.C) {
```

Comment:
Ah, we are replacing not found with meta not found, because this is a meta endpoint. Ok then, even if more descriptive messages would have been great.